### PR TITLE
Improve default metric formatting for RichProgressBar

### LINKED ISSUE #18367: Customizable Metric Formatting for Rich Progress Bar
### Description & Motivation

Allow users to customize the formatting of metric values in the Rich Progress Bar.

#### Background:
Currently, PyTorch Lightning's Rich Progress Bar formats all float values using the approach: `round(value, 3)`. While this approach works well for many common scenarios, it becomes ineffective for certain edge cases where the loss values are either too large or too small, rendering the displayed values not informative or potentially misleading.

#### Proposal:
Introduce an option to specify a custom formatting string for metrics displayed in the Rich Progress Bar. This would provide flexibility and cater to a broader range of use cases.

#### Benefits:
* Versatility: Users working with metrics in atypical ranges would be able to display them more effectively.
* Clarity: A custom format might help users present metric values in a more understandable or meaningful manner, aiding quick insights during training.
* Backward Compatibility: By keeping the default as round(value, 3), existing users won't see any changes unless they opt to use the new feature.

#### Suggested Implementation:

1) Add an optional parameter, say metric_format, to the Rich Progress Bar's initialization. This parameter would accept Python string format specifications.
2) Use this format string to display metric values if provided; otherwise, default to the existing behavior.

Optionally, I would like to see another optional flag named scientific_notation. When set to True, it should override the metric_format value and set it to ".3e" for displaying metrics in scientific notation format. If the user provides both the metric_format and sets scientific_notation to True, a warning should be issued informing the user that the scientific_notation flag will take precedence. This flag would increase usability because the scientific notation would be the most common custom formatting that a user needs to specify.

### Alternatives

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by quintenroets**
## What does this PR do?

Fixes #18367

This PR changes the default metric formatting from round(., 3) to using a ".3f" format string in "MetricsTextColumn" class, as discussed in https://github.com/Lightning-AI/lightning/pull/18373#discussion_r1309321063.

The PR comes with a unit test that tests the configurability of the metrics theme and a unit test that verifies the metric formatting behavior.

<details>
  <summary><b>Before submitting</b></summary>

- Was this **discussed/agreed** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/Lightning-AI/lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- Did you make sure to **update the documentation** with your changes? (if necessary)
- Did you write any **new necessary tests**? (not for typos and docs)
- [x] Did you verify new and **existing tests pass** locally with your changes?
- Did you list all the **breaking changes** introduced by this pull request?
- Did you **update the CHANGELOG**? (not for typos, docs, test updates, or minor internal changes/refactors)

</details>

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing, make sure you have read the [review guidelines](https://github.com/Lightning-AI/lightning/wiki/Review-guidelines). In short, see the following bullet-list:

<details>
  <summary>Reviewer checklist</summary>

- [ ] Is this pull request ready for review? (if not, please submit in draft mode)
- [ ] Check that all items from **Before submitting** are resolved
- [ ] Make sure the title is self-explanatory and the description concisely explains the PR
- [ ] Add labels and milestones (and optionally projects) to the PR so it can be classified

</details>

----
:books: Documentation preview :books:: https://pytorch-lightning--18483.org.readthedocs.build/en/18483/

### CODE REVIEW

File: src/lightning/pytorch/CHANGELOG.md [hunk 1]
```diff
@@ -9,6 +9,8 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/).
 
 ### Added
 
+- Changed default metric formatting from round(., 3) to ".3f" format string in "MetricsTextColumn" class ([#18483])(https://github.com/Lightning-AI/lightning/pull/18483)
```
Comment (carmocca) [inline src/lightning/pytorch/CHANGELOG.md:12]:
```suggestion
- Changed default metric formatting from `round(..., 3)` to `".3f"` format string in `MetricsTextColumn` class ([#18483])(https://github.com/Lightning-AI/lightning/pull/18483)
```

**Review by carmocca — APPROVED** [review]
LGTM

**Review by awaelchli — APPROVED** [review]
